### ISSUE TITLE
Add Filter support

### ISSUE THREAD

**Issue by Julias0** [Julias0/CapacitorSmsInboxReader#1]
I think the filter should contain:
- SMS received date ranges.
- Sender phone number.
- SIM (if a phone has multiple SIMs)

credits to @tewnut

**Comment by ganshul34** [comment]
If  I am not wrong is this  **_filter_** based on whether an **SMS** is **`Transactional`**, **`OTP`**, or **`Promotional`**?

**Comment by Julias0** [comment]
The filter may be anything - greater than a certain fate, a category filter like the one you mentioned etc. 

Right now this plugin gets all the messages out of the inbox but doesn't allow us to query the inbox of the device based on a certain condition. This can cause performance issues down the line

**Comment by ganshul34** [comment]
I am getting the error while running on `android ` (the custom plugin `SmsInboxReader` is not defined ).
Also, I created a standalone ionic-react app for testing purposes for this plugin.

**Comment by Julias0** [comment]
Can you try these two steps - 
 - Start a gradle sync in Android studio
 - inside MainActivity.java import SmsinboxReader and add it (there will be instructions in this file)
 
 If this dosnt work, can you add a screenshot of MainActivity.java

**Comment by ganshul34** [comment]
Yes, I need to import and add the `SmsinboxReader` in my `MainActivity.java`  file

**Comment by tewnut** [comment]
I think the filter should contain:
- SMS received date ranges.
- Sender phone number.
- SIM (if a phone has multiple SIMs)
